CLID-35: introduce filtering operators by selecting specific bundles (#824)
openshift/oc-mirror: docs/design/v2/operator-filtering-study.md

:warning: In this document, we do not discuss if/how the catalog would be rebuilt with the filtered content. We are separating the two discussions, even though we are aware that according to the solution chosen here, the filtered FBC catalog might not be considered as valid from the standpoint of OLM, or might not allow an operator to be upgraded correctly with such a filtered FBC catalog.

### Solutions implemented

TLDR; From the following studied solutions, S1 and S4(partial) are implemented for oc-mirror V2. 

> The user can filter operators and bundles by operator names, channels, minVersion/maxVersion as in V1. They can also list  specific bundles that they require for mirroring in the imageSetConfig.

Reviewer: It makes clear which one was implemented. Thanks for adding it.

It can be looked at as the longer term definitive solution.

> In 4.16, the API change to ImageSetConfig is implemented, allowing the user to list directly the bundles he/she needs to be mirrored. ([example](../../examples/imageset-config-filter-catalog-by-bundle.yaml))

Reviewer: ```suggestion
In 4.16, the API change to ImageSetConfig is implemented, allowing the user to list directly the bundles which are needed needs to be mirrored. ([example](../../examples/imageset-config-filter-catalog-by-bundle.yaml))
```
